File: services/academic-service/src/main/java/com/visor/school/academic/model/Schedule.java
package com.visor.school.academic.model;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.DayOfWeek;
import java.time.Duration;
import java.time.LocalTime;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Schedule value object for class scheduling
 */
@Embeddable
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Schedule {
    private String daysOfWeek; // Comma-separated days (e.g., "MONDAY,WEDNESDAY,FRIDAY")
    private LocalTime startTime;
    private LocalTime endTime;
    private String room;

    public List<DayOfWeek> getDays() {
        return Arrays.stream(daysOfWeek.split(","))
            .map(day -> {
                try {
                    return DayOfWeek.valueOf(day.trim().toUpperCase());
                } catch (IllegalArgumentException e) {
                    return null;
                }
            })
            .filter(day -> day != null)
            .collect(Collectors.toList());
    }

    public int getDurationMinutes() {
        return (int) Duration.between(startTime, endTime).toMinutes();
    }
}

File: services/academic-service/src/main/kotlin/com/visor/school/academicservice/model/Schedule.kt
package com.visor.school.academicservice.model

import jakarta.persistence.Embeddable
import java.time.DayOfWeek
import java.time.LocalTime

/**
 * Schedule value object for class scheduling
 */
@Embeddable
data class Schedule(
    val daysOfWeek: String, // Comma-separated days (e.g., "MONDAY,WEDNESDAY,FRIDAY")
    val startTime: LocalTime,
    val endTime: LocalTime,
    val room: String? = null
) {
    fun getDays(): List<DayOfWeek> {
        return daysOfWeek.split(",").mapNotNull { day ->
            try {
                DayOfWeek.valueOf(day.trim().uppercase())
            } catch (e: IllegalArgumentException) {
                null
            }
        }
    }

    fun getDurationMinutes(): Int {
        return java.time.Duration.between(startTime, endTime).toMinutes().toInt()
    }
}

